filters/openpolicyagent/authorizewithregopolicy/authorizewithregopolicy_test.go

```diff
@@ -0,0 +1,230 @@
+			clientServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Write([]byte("Welcome!"))
+				assert.True(t, isHeadersPresent(ti.backendHeaders, r.Header), "Enriched request header is absent.")
+				assert.True(t, isHeadersAbsent(ti.removeHeaders, r.Header), "Unwanted HTTP Headers present.")
+			}))
+
+			opaControlPlane := opasdktest.MustNewServer(
+				opasdktest.MockBundle("/bundles/"+ti.bundleName, map[string]string{
+					"main.rego": `
+						package envoy.authz
+
+						default allow = false
+
+						allow {
+							input.parsed_path = [ "allow" ]
+						}
+	
+						default allow_object = {
+							"allowed": false,
+							"headers": {"x-ext-auth-allow": "no"},
+							"body": "Unauthorized Request",
+							"http_status": 401
+						}
+						  
+						allow_object = response {
+							input.parsed_path = [ "allow", "structured" ]
+							response := {
+								"allowed": true,
+								"headers": {
+									"x-consumer": "x-consumer header value"
+								},
+								"request_headers_to_remove" : [
+									"x-remove-me",
+									"absent-header"
+								]
+							}
+						}
+					`,
+				}),
+			)
+
+			var routeFilters []*eskip.Filter
+			fr := make(filters.Registry)
+
+			config := []byte(fmt.Sprintf(`{
+				"services": {
+					"test": {
+						"url": %q
+					}
+				},
+				"bundles": {
+					"test": {
+						"resource": "/bundles/{{ .bundlename }}"
+					}
+				},
+				"plugins": {
+					"envoy_ext_authz_grpc": {    
+						"path": %q,
+						"dry-run": false    
+					}
+				}
+			}`, opaControlPlane.URL(), ti.regoQuery))
+
+			opaFactory := openpolicyagent.NewOpenPolicyAgentFactory()
+			ftSpec := NewAuthorizeWithRegoPolicySpec(opaFactory, openpolicyagent.WithConfigTemplate(config))
+			filterArgs := []interface{}{ti.bundleName}
+			_, err := ftSpec.CreateFilter(filterArgs)
+			if err != nil {
+				t.Fatalf("error in creating filter: %v", err)
+			}
+			fr.Register(ftSpec)
+			routeFilters = append(routeFilters, &eskip.Filter{Name: ftSpec.Name(), Args: filterArgs})
+
+			r := &eskip.Route{Filters: routeFilters, Backend: clientServer.URL}
+
+			proxy := proxytest.New(fr, r)
+			reqURL, err := url.Parse(proxy.URL)
+			if err != nil {
+				t.Errorf("Failed to parse url %s: %v", proxy.URL, err)
+			}
+			reqURL.Path = path.Join(reqURL.Path, ti.requestPath)
+			req, err := http.NewRequest("GET", reqURL.String(), nil)
+			for name, values := range ti.removeHeaders {
+				req.Header.Add(name, values[0]) //adding the headers to validate removal.
+			}
+
+			if err != nil {
+				t.Error(err)
+				return
+			}
+
+			rsp, err := http.DefaultClient.Do(req)
+			if err != nil {
+				t.Error(err)
+			}
+
+			assert.Equal(t, ti.expectedStatus, rsp.StatusCode, "HTTP status does not match")
+
+			assert.True(t, isHeadersPresent(ti.expectedHeaders, rsp.Header), "HTTP Headers do not match")
+
+			defer rsp.Body.Close()
+			body, err := io.ReadAll(rsp.Body)
+			if err != nil {
+				t.Error(err)
```

Comment:
In most cases you can use `t.Fatal` (fail quickly) instead of continue running the test with `t.Error`.